common/bindfstools.py

```diff
@@ -0,0 +1,157 @@
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation; either version 2 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License along
+#    with this program; if not, write to the Free Software Foundation, Inc.,
+#    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+import grp
+import subprocess
+import gettext
+
+import config
+import mount
+import logger
+import tools
+from exceptions import MountException
+
+_=gettext.gettext
+
+# TODO - quite a few shared methods between the various mount.MountControl subclasses. Perhaps move some common
+# methods into the base class?
+class Local(mount.MountControl):
+    """
+    Mount local path with bindfs. This allows for read-only viewing of snapshots stored on a locally mounted filesystem.
+    """
+
+    CHECK_FUSE_GROUP = False
+
+    def __init__(self, cfg = None, profile_id = None, hash_id = None, tmp_mount = False, parent = None, symlink = True, read_only = True, **kwargs):
+        self.config = cfg
+        if self.config is None:
+            self.config = config.Config()
+
+        self.profile_id = profile_id
+        if self.profile_id is None:
+            self.profile_id = self.config.get_current_profile()
+
+        self.tmp_mount = tmp_mount
+        self.hash_id = hash_id
+        self.parent = parent
+        self.symlink = symlink
+        self.read_only = read_only
+
+        #init MountControl
+        super(Local, self).__init__()
+
+        self.all_kwargs = {}
+
+        #First we need to map the settings.
+        self.setattr_kwargs('mode', self.config.get_snapshots_mode(self.profile_id), **kwargs)
+        self.setattr_kwargs('hash_collision', self.config.get_hash_collision(), **kwargs)
+        self.setattr_kwargs('path', self.config.get_local_path(self.profile_id), **kwargs)
+        self.setattr_kwargs('read_only', self.read_only, **kwargs)
+
+        self.set_default_args()
+
+        self.symlink_subfolder = None
+
+        self.log_command = '%s: %s' % (self.mode, self.path)
+
+    def _mount(self):
+        """
+        mount the local filesystem over bind
+        """
+        bindfs = ['bindfs', '-n']
+
+        # use read only mount if requested
+        if self.read_only:
+            bindfs.extend(['-o', 'ro'])
+
+        bindfs.extend([self.path, self.mountpoint])
+        #bugfix: bindfs doesn't mount if locale in LC_ALL is not available on remote host
+        #LANG or other envirnoment variable are no problem.
+        env = os.environ.copy()
+        if 'LC_ALL' in list(env.keys()):
+            env['LC_ALL'] = 'C'
```

Comment:
This was just a workaround for `sshfs`. I don't think `bindfs` has problems with locales, does it?